nengo/dists.py

```diff
@@ -192,17 +192,22 @@ class UniformHypersphere(Distribution):
     surface : bool
         Whether sample points should be distributed uniformly
         over the surface of the hyperphere (True),
-        or within the hypersphere (False).
-        Default: False
+        or within the hypersphere (False). Defaults to False.
+    min_magnitude : Number
+        Lower bound on the returned vector magnitudes (i.e.
+        magnitude >= min_magnitude). Must be in the range [0, 1]; defaults
+        to 0. Values outside this range will be clipped to this range. Ignored
+        if ``surface == True``
 
     """
 
-    def __init__(self, surface=False):
+    def __init__(self, surface=False, min_magnitude=0):
+        self.min_magnitude = np.clip(min_magnitude, 0, 1)
         self.surface = surface
 
     def __repr__(self):
-        return "UniformHypersphere(%s)" % (
-            "surface=True" if self.surface else "")
+        return "UniformHypersphere(min_magnitude=%s%r)" % (
+            ", surface=True" if self.surface else "", self.min_magnitude,)
```

Comment:
This is still not right. Also, min_magnitude should only appear if it's different from the default.